src/logic/hooks/useCanTxExecute.tsx

```diff
@@ -5,11 +5,21 @@ type UseCanTxExecuteType = (
   preApprovingOwner?: string,
   txConfirmations?: number,
   existingTxThreshold?: number,
+  txNonce?: string,
 ) => boolean
 
-const useCanTxExecute: UseCanTxExecuteType = (preApprovingOwner = '', txConfirmations = 0, existingTxThreshold) => {
+const useCanTxExecute: UseCanTxExecuteType = (
+  preApprovingOwner = '',
+  txConfirmations = 0,
+  existingTxThreshold,
+  txNonce,
+) => {
   const safeInfo = useSelector(currentSafe)
 
+  if (txNonce && parseInt(txNonce, 10) !== safeInfo.nonce) {
```

Comment:
`safeInfo.nonce` isn't a reliable way to check this.
Ideally, we should use the "recommended nonce" from the backend. See the `getRecommentedNonce` function.

Reply:
Also, please check that the edge case when nonce is 0 (first tx in Safe) still works.

Reply:
Tested the edge case also with a second queued tx with nonce 1 and it worked as expected. For the comparison we need the next executable nonce to decide whether the tx can be executed or not but `recommendedNonce` only returns nonces that are different than the next executable nonce.

Reply:
Thanks!